src/EditorFeatures/CSharpTest/Diagnostics/SpellCheck/SpellCheckTests.cs

```diff
@@ -461,5 +461,49 @@ void M()
 
             await TestAsync(text, expected, compareTokens: false);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsSpellcheck)]
+        [WorkItem(13345, "https://github.com/dotnet/roslyn/issues/13345")]
+        public async Task TestNotMissingOnKeywordWhichIsAlsoASnippet()
+        {
+            await TestAsync(
+@"
+class C
+{
+    void M()
+    {
+        // here 'for' is a snippet, and we should not offer it.
```

Comment:
Yup, you'r totally right.  I've tried ot make clear comments to indicate what's going on.